include/maths/CKdTree.h

```diff
@@ -181,6 +181,7 @@ class CKdTree {
     //! \note The \p points are moved into place.
     void build(TPointVec&& points) {
         this->build(points.begin(), points.end(), True());
+        points.clear();
```

Comment:
@dimitris-athanasiou, this is referring explicitly to an omission in the standard so that, for example, iterators which are held for a container which is subsequently moved remain valid, i.e. this is saying that the following is well formed:
```
std::vector<int> foo{1, 2, 3};
auto i = foo.begin();
std::vector<b> bar(std::move(foo));
*i = 2;
std::cout << bar[0] << std::endl;
```
output: 2. One is perfectly at liberty to call functions on rvalue references.